bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/persistence/PersistenceResource.java

```diff
@@ -201,41 +283,187 @@ private Response getItemHistoryBean(String serviceName, String itemName, String
+        filter.setBeginDate(dateTimeBegin);
+        filter.setEndDate(dateTimeEnd);
+        filter.setItemName(itemName);
+        try {
+            mService.remove(filter);
+        } catch (InvalidParameterException e) {
+            return JSONResponse.createErrorResponse(Status.BAD_REQUEST, "Invalid filter parameters.");
+        }
+
+        return JSONResponse.createResponse(Status.OK, null, "");
+    }
+
+    private Response putItemState(String serviceName, String itemName, String value, String time) {
+        // If serviceName is null, then use the default service
+        PersistenceService service = null;
+        if (serviceName == null) {
+            serviceName = PersistenceExtensions.getDefaultService();
+        }
+        service = persistenceServices.get(serviceName);
+
+        if (service == null) {
+            logger.warn("Persistence service not found '{}'.", serviceName);
+            return JSONResponse.createErrorResponse(Status.BAD_REQUEST,
+                    "Persistence service not found: " + serviceName);
+        }
+
+        Item item;
+        try {
+            if (itemRegistry == null) {
+                logger.warn("Item registry not set.");
+                return JSONResponse.createErrorResponse(Status.CONFLICT, "Item registry not set.");
+            }
+            item = itemRegistry.getItem(itemName);
+        } catch (ItemNotFoundException e) {
+            logger.warn("Item not found '{}'.", itemName);
+            return JSONResponse.createErrorResponse(Status.BAD_REQUEST, "Item not found: " + itemName);
+        }
+
+        // Try to parse a State from the input
+        State state = TypeParser.parseState(item.getAcceptedDataTypes(), value);
+        if (state == null) {
+            // State could not be parsed
+            logger.warn("Can't persist item {} with invalid state '{}'.", itemName, value);
+            return JSONResponse.createErrorResponse(Status.BAD_REQUEST, "State could not be parsed: " + value);
+        }
+
+        Date dateTime = null;
+        if (time != null && time.length() != 0) {
+            dateTime = convertTime(time);
+        }
+        if (dateTime == null || dateTime.getTime() == 0) {
+            logger.warn("Error with persistence store to {}. Time badly formatted {}.", itemName, time);
+            return JSONResponse.createErrorResponse(Status.BAD_REQUEST, "Time badly formatted.");
+        }
+
+        if (!(service instanceof ModifiablePersistenceService)) {
+            logger.warn("Persistence service not modifiable '{}'.", serviceName);
+            return JSONResponse.createErrorResponse(Status.BAD_REQUEST,
+                    "Persistence service not modifiable: " + serviceName);
+        }
+
+        ModifiablePersistenceService mService = (ModifiablePersistenceService) service;
 
-        return beanList;
+        mService.store(item, dateTime, state);
+        return JSONResponse.createResponse(Status.OK, "", "");
```

Comment:
if you do not have a message to tell, you don't need to construct a JSONResponse. Use `Response.status(Status.OK).build();` instead.